### DISCUSSION TITLE
use spi1 and spi2

### DISCUSSION THREAD

**Discussion by hhl-16** [stm32duino/Arduino_Core_STM32#1904] [Q&A] [answered]
hi guys 
can i use spi1 and spi2 for 2 sensors with STM32F407VE
spi1 for dwm1000
sp2 for module micro sd
thanks

**Comment by fpistm** [comment, marked as answer]
Hi,
yes you can. or even use the same SPI instance with a different CS.
SATM32F407VE has 3 SPI peripherals:
https://github.com/stm32duino/Arduino_Core_STM32/blob/b57fab977026c70883fd22a8a4853334c7ee20ff/variants/STM32F4xx/F407V(E-G)T_F417V(E-G)T/PeripheralPins.c#L248-L292

By default only one instance is defined named `SPI`, depending of the variant selected, for the generic:
https://github.com/stm32duino/Arduino_Core_STM32/blob/b57fab977026c70883fd22a8a4853334c7ee20ff/variants/STM32F4xx/F407V(E-G)T_F417V(E-G)T/variant_generic.h#L173-L181

Which means it uses `SPI1`.

For other instance you have to define it:
https://github.com/stm32duino/wiki/wiki/API#spi
